src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,511 @@
+loop’s code block *always runs at least once*. At the end of the block’s
+execution, we check a condition to determine if we should run the block
+again. This nicely mimics the behavior of simple menu-driven
+applications.
+
+Within this loop, menu options are printed to the screen and user input
+is collected. If the input is valid, it returns the choice as a String
+to the caller. This String corresponds to the chosen key (from
+``choices``, which will be either ``actionChoices`` or
+``columnChoices``) of the item the user selected. If invalid, it
+re-prompts the user.
+
+The local variable ``choiceKeys`` is used to easily enumerate the
+``choices`` ``HashMap``. In other words, it gives us a simple way to
+provide an ordering to ``choices``, which doesn’t have an ordering of
+its own.
+
+The JobData Class
+------------------
+
+The ``JobData`` class is responsible for importing the data from the CSV
+file and parsing it into a Java-friendly format, that is, into
+``HashMap`` and ``ArrayList`` form. Look toward the bottom of the class
+and you will see a method named ``loadData``, which does just what it
+advertises. After parsing the file data, it stores the data in the
+private property ``allJobs`` which is of type
+``ArrayList<HashMap<String, String>>``.
+
+.. admonition:: Note
+
+   We haven’t covered static properties and methods in-depth yet. For this
+   assignment, know simply that they allow us to use properties and methods
+   of a class without creating an object from that class. For example, we
+   can call ``JobData.findAll()`` from the ``TechJob`` class.
+
+   If you want to create a new method in ``JobData``, or add a property, be
+   sure to declare it as ``static``.
+
+Let’s look more closely at the data type of ``allJobs``. It purports to
+be an ``ArrayList`` that stores ``HashMap`` objects which have
+``String`` keys and ``String`` values. If we were to represent some of
+this data visually, using ``[]`` for an ``ArrayList`` and ``{}`` with
+key/value pairs, it would look like this:
+
+.. sourcecode:: java
+   :linenos:
+
+   [
+       {
+           "name": "Junior Data Analyst",
+           "employer": "Lockerdome",
+           "location": "Saint Louis",
+           "position type": "Data Scientist / Business Intelligence",
+           "core competency": "Statistical Analysis"
+       },
+       {
+           "name": "Junior Web Developer",
+           "employer": "Cozy",
+           "location": "Portland",
+           "position type": "Web - Back End",
+           "core competency": "Ruby"
+       },
+       ...
+   ]
+
+If you look at the ``loadData`` method you’ll see a lot of unfamiliar code.
+Blake wrote this essential piece of code for you, and while you won’t have to
+modify it, it will be useful to have an idea of how it works. Read
+through the code until you feel like you can describe its functionality
+at a basic level.
+
+.. index:: ! overloading
```

Comment:
The classes content will talk about overloading. Perhaps this should be a secondary index reference.

Reply:
Switched to secondary reference.